redgenes/gene_annotations.py

```diff
@@ -370,14 +382,23 @@ def annotation_pipeline(df, tmpdir, kofamscan_profile, kofamscan_kolist):
 def main():
```

Comment:
```suggestion
def _unlink_directory(path):
    def f():
        shutil.rmtree(path)
    return f
    
    
@click.command()
@click.option('--metadata', type=click.Path(exists=True), required=True)
@click.option('--ko-profile', type=click.Path(exists=True), required=True)
@click.option('--ko-kolist', type=click.Path(exists=True), required=True)
@click.option('--working-dir', type=click.Path(exists=True), required=False) 
def main(metadata, ko_profile, ko_kolist, working_dir):
    if working_dir is None:
        working_dir = tempfile.TemporaryDirectory()
        atexit.register(_unlink_directory(working_dir))
```

...the working dir piece allows the caller (ie the submission script) to setup a working dir and manage it separately. Otherwise, this uses the formal semantics to construct a temp directory (which will be under `TMPDIR` if set in the environment), and we clean up at exit